wafer/schedule/admin.py

```diff
@@ -169,6 +169,26 @@ def check_schedule():
     return True
 
 
+@cache_result('wafer_schedule_validate_schedule', 30)
+def validate_schedule():
+    """Helper routine to easily test if the schedule is valid"""
+    all_items = prefetch_schedule_items()
+    errors = []
+    if find_clashes(all_items):
+        errors.append('Clashes found in schedule.')
+    if find_duplicate_schedule_items(all_items):
+        errors.append('Duplicate schedule items found in schedule.')
+    if validate_items(all_items):
+        errors.append('Invalid schedule items found in schedule.')
+    if find_overlapping_slots():
+        errors.append('Overlapping slots found in schedule.')
+    if find_non_contiguous(all_items):
+        errors.append('Non contiguous slots found in schedule.')
+    if find_invalid_venues(all_items):
+        errors.append('Invalid venues found in schedule.')
+    return errors
+
+
```

Comment:
There's benefit to consolidating valiate_schedule and check_schedule, but we shouldn't blindly replace one with the other. check_schedule is called when rendering the schedule, so we want to cache it's result for a long time as the schedule view is accessed fairly often during the conference. This is quite different from the caching behaviour we want for validation during editing, when responsiveness is more of an issue abnd it's not being accessed by many people at once.